activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb

```diff
@@ -171,6 +173,10 @@ def supports_explain?
         true
       end
 
+      def supports_foreign_keys?
+        true
+      end
+
```

Comment:
I'll have it do the check and cache the result when it enables foreign keys on connection start.